core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -283,14 +284,21 @@ public boolean removeKey(ECKey key) {
         }
     }
 
+    /**
+     * Returns the earliest creation time of keys in this chain.
+     * @return earliest creation times of keys in this chain,
+     *         {@link Instant#EPOCH} if at least one time is unknown,
+     *         {@link Instant#MAX} if no keys in this chain
+     */
     @Override
-    public long getEarliestKeyCreationTime() {
+    public Instant getEarliestKeyCreationTimeInstant() {
         lock.lock();
         try {
             return hashToKeys.values().stream()
                     .mapToLong(ECKey::getCreationTimeSeconds)
-                    .min()
-                    .orElse(Long.MAX_VALUE);
+                    .mapToObj(t -> Instant.ofEpochSecond(t))
```

Comment:
You'll get an exception here if `t` (time in seconds) is `Long.MAX_VALUE`. You might need a mapping function that maps `Long.MAX_VALUE` to `Instant.MAX`. I can't remember where in the code I ran into this issue, but be careful.

Reply:
Good point. Would

```
mapToObj(t -> t == Long.MAX_VALUE ? Instant.MAX : Instant.ofEpochSecond(t))
```

work?

Reply:
I just noticed I have to do the Long.MAX_VALUE <> Instant.MAX conversion at other places, too.

Reply:
> Good point. Would
> 
> ```
> mapToObj(t -> t == Long.MAX_VALUE ? Instant.MAX : Instant.ofEpochSecond(t))
> ```

Yes, that should work. I would write a utility method for this so you can use a method reference.

Reply:
Well, that piece of code will be gone soon, as `getCreationTime()` will migrate to `Instant` too. Coming up in this theatre…